components/prod/lncs-hmcs.tf

```diff
@@ -0,0 +1,15 @@
+data "local_file" "lncs-hmcs-configuration" {
+  filename = "${path.cwd}/../../environments/prod/lncs-hmcs.yml"
+}
+
+module "lncs-hmcs" {
+  source              = "../../modules/azure-private-dns/"
+  cname_records       = yamldecode(data.local_file.reform-configuration.content).cname
```

Comment:
this should be
`cname_records       = yamldecode(data.local_file.lncs-hmcs-configuration.content).cname`